### ISSUE TITLE
typeUnparen: reports false-positive `(func())(nil)`

### ISSUE THREAD

**Issue by tdakkota** [go-critic/go-critic#1245]
`typeUnparen` reports `could simplify (func()) to func()`, but `func() (nil)` is invalid function literal and definitely is not the same expression.

Reproducer:
```go
package yaml

func _() {
	_ = (func())(nil)
}
```

Discovered [here](https://github.com/tdakkota/yamlx/commit/8b278946d3a6619f6493e1a7423f5d3f05dcc9b8#diff-5afd41e7e0a70c3a6f0e1572f6938d9aacf5bc590033f7e798a0fb6f741e51ddR42).

`go-critic` version (I've installed it using `go install`, so there is no `gocritic version` information):
```
$ go version -m (where.exe gocritic)
C:\Users\tdakkota\go\bin\gocritic.exe: go1.19
        mod     github.com/go-critic/go-critic  v0.6.3  h1:abibh5XYBTASawfTQ0rA7dVtQT+6KzpGqb/J+DxRDaw=
        ...
```

The `golangci-lint` version has the same issue.
